### ISSUE TITLE
Bug in Relationship dialog - incompatible data types

### ISSUE THREAD

**Issue by edhans** [TabularEditor/TabularEditor3#855]
Regardless of the data types, the Create New Relationship dialog box reports the columns are of a different type. It will let me click OK and create the relationship, but it is still wrong info. You can see both of these are DateTime datatypes. I've repro'd this with strings too.

This remains an "error" if you save it, then open it back up to edit it.

There is no error in the Relationship view of the TOM panel. 

![image](https://user-images.githubusercontent.com/45298358/226450677-5a4dd2ac-fc55-4b2c-8808-0968574f69c7.png)
![image](https://user-images.githubusercontent.com/45298358/226450707-daf0f651-a7b9-4ec2-9452-5fc8488d0c4d.png)

**Comment by mlonsk** [comment]
Hi @edhans 
I have reproduced this bug and we will look into fixing it. 
Thank you for reporting.

**Comment by mlonsk** [comment]
Hi @redstor-albe 
The issue that is being reported here is according to our testing only a display issue, ie. TE is showing incompatible datatypes though they are the same. It should not affect the actual relationship being created. 

Are you able to provide is with an example of what the relationship is you are setting up, properties etc and the error you are seeing in Power BI?
Does the same issue occur if you create or modify a different relationship?

**Comment by redstor-albe** [comment]
Hi @mlonsk 

After posting this message yesterday I have since downgraded to 3.4.2 - my problem still persisted so I deleted the relationships and recreated them, that seems to have solved the problem. One case I had was one to many between my date table and a fact table containing some data and dates, the other was a 1:1 relationship using GUIDS. All stopped working but after recreating them and refreshing the model everything worked.  
Don't think it was related to this bug.  
By the way it was definitely in the model (AAS hosted) - I could see the relationships not working in PBI, Dax Studio and Tabular Editor.
